.github/actions/bin/create_image_table.py

```diff
@@ -40,46 +40,58 @@ def stage_sort(item):
 
 # File written by a previous workflow step
 with open(f"{github_workspace}/built_images.json") as bij:
-  msg(f"Reading image build data from {bij.name}:")
-  data = []
-  for build in json.load(bij):  # list of build data maps
-    stage = build.get("stage", False)
-    name = build.get("name", False)
-    sfx = build.get("sfx", False)
-    task = build.get("task", False)
-    if bool(stage) and bool(name) and bool(sfx) and bool(task):
-        image_suffix = f'{stage[0]}{sfx}'
-        data.append(dict(stage=stage, name=name,
-                         image_suffix=image_suffix, task=task))
-        if cirrus_ci_build_id is None:
-            cirrus_ci_build_id = sfx
-        msg(f"Including '{stage}' stage build '{name}' for task '{task}'.")
-    else:
-        msg(f"Skipping  '{stage}' stage build '{name}' for task '{task}'.")
-
-url = 'https://cirrus-ci.com/task'
+    msg(f"Reading image build data from {bij.name}:")
+    data = []
+    for build in json.load(bij):  # list of build data maps
+        stage = build.get("stage", False)
+        name = build.get("name", False)
+        sfx = build.get("sfx", False)
+        task = build.get("task", False)
+        if bool(stage) and bool(name) and bool(sfx) and bool(task):
+            image_suffix = f"{stage[0]}{sfx}"
+            data.append(
+                dict(stage=stage, name=name, image_suffix=image_suffix, task=task)
+            )
+            if cirrus_ci_build_id is None:
+                cirrus_ci_build_id = sfx
+            msg(f"Including '{stage}' stage build '{name}' for task '{task}'.")
+        else:
+            msg(f"Skipping  '{stage}' stage build '{name}' for task '{task}'.")
+
+url = "https://cirrus-ci.com/task"
 lines = []
 data.sort(key=stage_sort)
 for item in data:
-  lines.append('|*{0}*|[{1}]({2})|`{3}`|\n'.format(item['stage'],
-    item['name'], '{0}/{1}'.format(url, item['task']),
-    item['image_suffix']))
+    image_suffix = item["image_suffix"]
+    # Base-images should never actually be used, but it may be helpful
+    # to have them in the list in case some debugging is needed.
+    if item["stage"] != "cache":
+        image_suffix = "do-not-use"
+    lines.append(
+        "|*{0}*|[{1}]({2})|`{3}`|\n".format(
+            item["stage"],
+            item["name"],
+            "{0}/{1}".format(url, item["task"]),
+            image_suffix,
+        )
```

Comment:
@edsantiago this is the main change, everything else is automated-style updates.  Does this wording seem okay?

Reply:
I think that makes sense, thanks. I'll wait to see the table though.

Quick plea: the order of commits (change, then reformat; instead of reformat, add new code) makes it hard to review. Next time, could you do cleanup first, then new code? Or add new code using the approved format? TIA.

Reply:
Aye, thanks.  I thought and struggled with that, deciding 'good enough' would be to make it a separate commit.  Though I trust your experience that the commit order matters too.  

Python lint/style tooling is annoying in that the standards seem to be a constantly moving target. Otherwise I'd have included it in the validation checks here :confused: